plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -4729,6 +4729,17 @@ private VirtualMachineGuestOsIdentifier translateGuestOsIdentifier(String cpuArc
         return VirtualMachineGuestOsIdentifier.OTHER_GUEST;
     }
 
+    private String translateGuestOsIdentifierEx(String cpuArchitecture, String guestOs, String cloudGuestOs) {
```

Comment:
Curious, why 2 or 3; not 10 or 100?

This method can be removed if PR https://github.com/apache/cloudstack/pull/1397 is accepted, which bumps vmware dependency version to sdk 6.0 which has the additional guest os identifier (i.e. no need to this extra translation).